lib/icinga/pluginutility.cpp

```diff
@@ -38,11 +39,29 @@ void PluginUtility::ExecuteCommand(const Command::Ptr& commandObj, const Checkab
 	const std::function<void(const Value& commandLine, const ProcessResult&)>& callback)
 {
 	Value raw_command = commandObj->GetCommandLine();
-	Dictionary::Ptr raw_arguments = commandObj->GetArguments();
+	Value raw_plugin = commandObj->GetPlugin();
 
 	Value command;
+	Dictionary::Ptr raw_arguments = commandObj->GetArguments();
 
 	try {
+		if (!raw_plugin.IsEmpty()) {
+			Value plugin = LookupPlugin(raw_plugin);
+
+			if (!plugin.IsEmpty()) {
```

Comment:
I'm not sure this block is a good idea. The `command` attribute might contain invocation-specific arguments (macros, functions).